### PR TITLE
MGMT-17227: Use openshift-install for creation of ibi-config.iso

### PR DESCRIPTION
**Pull request by javipolo**

### CODE REVIEW

File: Makefile.ibi [hunk 1]
```diff
@@ -121,11 +146,51 @@ $(IBI_CLUSTER_CONFIG_PATH): ibi-certs credentials/pull-secret.json
...
+	sudo cp $@ $(LIBVIRT_IMAGE_PATH)/ibi-config.iso
+
 .PHONY: ibi-config
 ibi-config: $(IBI_CLUSTER_CONFIG_PATH) $(IBI_CLUSTER_CONFIG_MANIFESTS)
 
 .PHONY: ibi-config.iso
-ibi-config.iso: ibi-config ## Create ibi-config.iso
+ibi-config.iso: ibi-config ## Create ibi-config.iso - DEPRECATED
 	mkisofs -o $@ -R -V "cluster-config" $(IBI_CONFIG_DIR)
 	sudo cp $@ $(LIBVIRT_IMAGE_PATH)
```
Comment (eranco74) [inline Makefile.ibi:195]: I think we want to delete this

Comment (eranco74) [reply Makefile.ibi:195]: We can also delete these other make targets:
ibi-config
ibi-certs
$(IBI_CLUSTER_CONFIG_PATH)
And this script: 
 https://github.com/rh-ecosystem-edge/ib-orchestrate-vm/blob/master/ibi/generate_certs.sh

File: Makefile.ibi [hunk 2]
```diff
@@ -121,11 +146,51 @@ $(IBI_CLUSTER_CONFIG_PATH): ibi-certs credentials/pull-secret.json
...
+		HOST_ROUTE=$(shell $(virsh) net-dumpxml $(IBI_NET_NAME) | grep '<ip ' | xargs -n1 | grep address | cut -d = -f 2) \
+		RELEASE_REGISTRY=$(IBI_RELEASE_REGISTRY) \
+		envsubst > $(IBI_WORK_DIR)/imagebased-config.yaml
+
+.PHONY: imagebasedconfig.iso
+imagebasedconfig.iso: RELEASE_VERSION=$(SEED_VERSION)
+imagebasedconfig.iso: $(IBI_WORK_DIR)/imagebasedconfig.iso ## Create imagebasedconfig.iso using openshift-installer
+
+.PHONY: $(IBI_WORK_DIR)/imagebasedconfig.iso
+$(IBI_WORK_DIR)/imagebasedconfig.iso: RELEASE_VERSION=$(SEED_VERSION)
```
Comment (eranco74) [inline Makefile.ibi:184]: Consider replacing the `sudo cp $@ $(LIBVIRT_IMAGE_PATH)/ibi-config.iso` with updating `ibi-attach-config.iso` to use `$(IBI_WORK_DIR)/imagebasedconfig.iso`

File: Makefile.ibi [hunk 3]
```diff
@@ -11,7 +12,21 @@ IBI_CLUSTER_NAME ?= $(IBI_VM_NAME)
 IBI_HOSTNAME ?= $(IBI_VM_NAME)
 IBI_INSTALLATION_DISK ?= /dev/vda
 IBI_WORK_DIR ?= ibi-iso-work-dir
-IBI_KUBECONFIG=kubeconfig.ibi
+IBI_INSTALL_CONFIG ?= $(IBI_WORK_DIR)/install-config.yaml
+IBI_KUBECONFIG = $(IBI_WORK_DIR)/auth/kubeconfig
+
+OPENSHIFT_INSTALLER_REPO ?= https://github.com/mresvanis/installer
+OPENSHIFT_INSTALLER_BRANCH ?= add-config-iso-cluster-configuration
```
[the remark this answers is not in the document]
Comment (eranco74) [reply Makefile.ibi:19]: OK, we can update once it's ready

**Comment by javipolo** [comment]
/hold

@jeff-roche says that this might break CI, so we want to test it in CI environment first

File: Makefile.ibi [hunk 4]
```diff
@@ -11,7 +12,23 @@ IBI_CLUSTER_NAME ?= $(IBI_VM_NAME)
 IBI_HOSTNAME ?= $(IBI_VM_NAME)
 IBI_INSTALLATION_DISK ?= /dev/vda
 IBI_WORK_DIR ?= ibi-iso-work-dir
-IBI_KUBECONFIG=kubeconfig.ibi
+IBI_INSTALL_CONFIG ?= $(IBI_WORK_DIR)/install-config.yaml
+IBI_KUBECONFIG = $(IBI_WORK_DIR)/auth/kubeconfig
+
+OPENSHIFT_INSTALLER_REPO ?= https://github.com/mresvanis/installer
```
Comment (omertuc) [inline Makefile.ibi:18]: Add a comment/TODO explaining why we're using this fork

Comment (javipolo) [reply Makefile.ibi:20]: Agree, comment added

File: Makefile.ibi (hunk 3 above)
[the remark this answers is not in the document]
Comment (mresvanis) [reply Makefile.ibi:21]: FYI [this](https://github.com/openshift/installer/pull/8514) PR is the one we should wait to get merged.

[the remark this answers is not in the document]
Comment (javipolo) [reply Makefile.ibi:20]: Thanks @mresvanis , added the PR to the comment in the Makefile :)

**Pull request merged**